tests/ui/associated-inherent-types/issue-111879-1.stderr

```diff
@@ -0,0 +1,12 @@
+error[E0580]: `main` function has wrong type
+  --> $DIR/issue-111879-1.rs:12:1
+   |
+LL | fn main(_: for<'a> fn(Foo<fn(&'a ())>::Assoc)) {}
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ incorrect number of function parameters
+   |
+   = note: expected fn pointer `fn()`
```

Comment:
Side-note, it would be nice if we got this to say "expected signature"... but the problem is that `require_same_types` takes (fn ptr) types instead of poly fn sigs. Whatever, I guess.